src/meta/processors/indexMan/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,97 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/ActiveHostsMan.h"
+#include "meta/processors/admin/AdminClient.h"
+#include "meta/processors/indexMan/RebuildEdgeIndexProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void RebuildEdgeIndexProcessor::process(const cpp2::RebuildIndexReq& req) {
+    auto space = req.get_space_id();
+    CHECK_SPACE_ID_AND_RETURN(space);
+    const auto &indexName = req.get_index_name();
+    LOG(INFO) << "Rebuild Edge Index Space " << space << ", Index Name " << indexName;
+
+    std::unique_ptr<AdminClient> client(new AdminClient(kvstore_));
+    auto partsRet = client->getLeaderDist(space).get();
+    if (!partsRet.ok()) {
+        LOG(ERROR) << "Get space " << space << "'s part failed";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    auto parts = partsRet.value();
+
+    folly::SharedMutex::ReadHolder schemaHolder(LockUtils::edgeLock());
```

Comment:
I want to know why we need two locks here.

Reply:
Missed. Because when schema have create index on it, it would not drop or alter. So we could just hold index lock, it is enough.